x-pack/plugins/observability_solution/dataset_quality/common/api_types.ts

```diff
@@ -16,6 +16,7 @@ export const dataStreamStatRt = rt.intersection([
     sizeBytes: rt.number,
     lastActivity: rt.number,
     integration: rt.string,
+    docsCount: rt.union([rt.null, rt.number]), // rt.null is only needed for https://github.com/elastic/kibana/issues/178954
```

Comment:
What about aligning this name with the one in `common/constants.ts`? so the two of them are either `docsCount` or `totalDocs`?